docs(weave): document PII redaction from traces feature in Python SDK (#3618)
wandb/weave: docs/docs/guides/tracking/redact-pii.md

# Redact PII from Traces

> Some organizations process Personally Identifiable Information (PII) such as names, phone numbers, and email addresses in their Large Language Model (LLM) workflows. Storing this data within Weave poses compliance and security risks.

Author: ```suggestion
Some organizations process Personally Identifiable Information (PII) such as names, phone numbers, and email addresses in their Large Language Model (LLM) workflows. Storing this data in Weights & Biases (W&B) Weave poses compliance and security risks.
```

> The W&B Weave _Sensitive Data Protection_ feature allows you to automatically redact Personally Identifiable Information (PII) from a [trace](../tracking/index.md) before it is sent to Weave servers. This feature integrates [Microsoft Presidio](https://microsoft.github.io/presidio/) into the Weave Python SDK, which means that you can control redaction settings at the SDK level.

Author: ```suggestion
The  _Sensitive Data Protection_ feature allows you to automatically redact Personally Identifiable Information (PII) from a [trace](../tracking/index.md) before it is sent to Weave servers. This feature integrates [Microsoft Presidio](https://microsoft.github.io/presidio/) into the Weave Python SDK, which means that you can control redaction settings at the SDK level.
```

:::important
This feature is only available for enterprise customers, and only available in the Python SDK.
:::

The Sensitive Data Protection feature introduces the following functionality to the Python SDK:

- A `redact_pii` setting, which can be toggled on or off in the `weave.init` call to enable PII redaction.

> - Automatic redaction of `PERSON`, `PHONE_NUMBER`, and `EMAIL_ADDRESS` when `redact_pii = True`.

Reviewer: This is a list of the default redacted fields:     "CREDIT_CARD",
    "CRYPTO",
    "EMAIL_ADDRESS",
    "IBAN_CODE",
    "IP_ADDRESS",
    "LOCATION",
    "PERSON",
    "PHONE_NUMBER",
    "US_SSN",
    "US_BANK_NUMBER",
    "US_DRIVER_LICENSE",
    "US_PASSPORT",
    "UK_NHS",
    "UK_NINO",
    "ES_NIF",
    "IN_AADHAAR",
    "IN_PAN",
    "FI_PERSONAL_IDENTITY_CODE",

Author: Updated in https://github.com/wandb/weave/pull/3618/commits/cd680a189eab3f606e3f7ecabcf1d2ef0a0830ef

-  Customizable redaction fields using the configurable `redact_pii_fields` setting.

## Enable PII redaction

To get started with the Sensitive Data Protection feature in Weave, complete the following steps:

1. Install the required dependencies:

    ```bash
    pip install presidio-analyzer presidio-anonymizer
    ```

2. Modify your `weave.init` call to enable redaction. By default, `PERSON`, `PHONE_NUMBER` and `EMAIL_ADDRESS` are redacted when `redact_pii=True`:

    ```python
    import weave

>     weave.init("my-project", redact_pii=True)

Reviewer: It would actually be this line
```
weave.init('weave-sentinel-test', settings={"redact_pii": True})
```

Author: Updated in https://github.com/wandb/weave/pull/3618/commits/cd680a189eab3f606e3f7ecabcf1d2ef0a0830ef

    ```

3. (Optional) Customize redaction fields using the `redact_pii_fields` parameter:

    ```python

>     weave.init("my-project", redact_pii=True, redact_pii_fields=["CREDIT_CARD", "SSN"])

Reviewer: The fields can be set like this.

```python
weave.init('weave-sentinel-test', settings={"redact_pii": True, "redact_pii_fields"=["CREDIT_CARD", "US_SSN"]})
```

Author: Got it, updated in https://github.com/wandb/weave/pull/3618/commits/cd680a189eab3f606e3f7ecabcf1d2ef0a0830ef

    ```

## Usage information 

> 

Reviewer: We should probably include a list, or atleast a link to the full list of fields that Presidio can redact. That can be found here: https://microsoft.github.io/presidio/supported_entities/

Author: Added a new section with the list of default redacted entities in https://github.com/wandb/weave/pull/3618/commits/cd680a189eab3f606e3f7ecabcf1d2ef0a0830ef and linked out to it where it made sense to

- Enabling redaction increases processing time due to the Presidio dependency.

> - If you don't have an enterprise subscription, you will receive an error if you try to enable this feature.

Reviewer: > you will receive an error if you try to enable this feature.

This actually currently is not true. This feature can be turned on by anyone with this release, but we will likely restrict it to enterprise in the future, so it's probably worth keeping in here for now.

Author: Got it. I don't think we should put any info that isn't true or is potentially misleading, so I've just removed anything that mentions this as an enterprise specific feature in https://github.com/wandb/weave/pull/3618/commits/cd680a189eab3f606e3f7ecabcf1d2ef0a0830ef. 

If / when it is restricted to enterprise, we can definitely update the docs to reflect that

# Redact PII from Traces

:::important

> This feature is only available in the Python SDK.

Author: ```suggestion
This feature is only available for Enterprise users, and is only accessible via the Python SDK.
```

> This feature is only accessible via the Python SDK.

Author: Re-revert of Enterprise only per discussion

```suggestion
This feature is only available for Enterprise users, and is only accessible via the Python SDK.
```

> This feature is only available for Enterprise users, and is only accessible via the Python SDK.

Author: ```suggestion
This feature is only accessible via the Python SDK.
```
